keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -402,182 +393,262 @@ The following new `AdminNetworkPolicy` API will be added to the `policy.networki
+
+  // The port on the given protocol. This can either be a numerical or named
+  // port on a pod. If this field is not provided, this matches no port names and
+  // numbers.
+  // If present, only traffic on the specified protocol AND port will be matched.
+  // +optional
+  Port       *intstr.IntOrString
+    
+  // If set, indicates that the range of ports from port to endPort, inclusive,
+  // should be allowed by the policy. This field cannot be defined if the port field
+  // is not defined or if the port field is defined as a named (string) port.
+  // The endPort must be equal or greater than port.
+  // This feature is in Beta state and is enabled by default.
+  // It can be disabled using the Feature Gate "NetworkPolicyEndPort".
   // +optional
-  To       []AdminNetworkPolicyPeer
+  EndPort     *int32 
 }
 
 const (
   // RuleActionPass enables admins to provide exceptions to ClusterNetworkPolicies and delegate this rule to
   // K8s NetworkPolicies.
-  RuleActionPass    RuleAction = "Pass"
+  AdminNetPolRuleActionPass    AdminNetPolRuleAction = "Pass"
   // RuleActionDeny enables admins to deny specific traffic.
-  RuleActionDeny    RuleAction = "Deny"
+  AdminNetPolRuleActionDeny    AdminNetPolRuleAction = "Deny"
   // RuleActionAllow enables admins to specifically allow certain traffic.
-  RuleActionAllow   RuleAction = "Allow"
+  AdminNetPolRuleActionAllow   AdminNetPolRuleAction = "Allow"
 )
 
-// Exactly one of the selector pointers should be set in a given AdminNetworkPolicyPeer
+// AdminNetworkPolicyPeer defines an in-cluster peer to allow traffic to/from. 
+// Exactly one of the selector pointers should be set for a given peer. 
 type AdminNetworkPolicyPeer struct {
-  NamespaceSetSelector         *NamespaceSet
-  NamespaceSetAndPodSelector   *NamespaceSetAndPod
+  Namespaces       *NamespaceSet
+  NamespacedPods   *NamespaceAndPodSet
 }
 
 // NamespaceSet defines a flexible way to select Namespaces in a cluster.
+// Exactly one of the selectors should be set.  If a consumer observes none of 
+// its fields are set, they should assume an option they are not aware of has 
+// been specified and fail closed.
 type NamespaceSet struct {
-  Scope               NamespaceMatchType
-  // Labels should be set only when scope is "SameLabels" or "NotSameLabels.
-  // +optional
-  Labels              []string
-  // NamespaceSelector should be set only when scope is "Selector". 
+  // Self cannot be "false" when it is set.  
+  // If Self is "true" then all pods in the subject's namespace are selected.
+  Self                *bool
+  // NotSelf cannot be "false" when it is set.  
+  // if NotSelf is "true" then all pods not in the subject's Namespace are selected.
+  NotSelf             *bool
+  // NamespaceSelector is a labelSelector used to select Namespaces, This field 
+  // follows standard label selector semantics; if present but empty, it selects 
+  // all Namespaces. 
   NamespaceSelector   *metav1.LabelSelector
```

Comment:
just `Selector`?  "Namespace" seems clear, no?

Edit: see comment in PodSet.  We don't have a mechanism for grouping namespaces, other that selectors, but what if we did?  E.g. what if we had a "tenant" concept.  Would we wish that all of these were named `SameNamespace` and `SameNamespaceLabels`, etc?  Or would we add a different option in the Peer, like "Tenant` and TenantPods`?

Don't need to block KEP on it, but worth considering, since we have talked about ServiceAccount for pods as an alternative to Selector.  Maybe the way to do that would be a magic "k8s.io/service-acount" label on pods...

Reply:
Left it the way it is for now, will revisit this